### DISCUSSION TITLE
Is it possible to generate .c4 file from parsed object model?

### DISCUSSION THREAD

**Discussion by dzharikhin** [likec4/likec4#914] [Q&A] [answered]
I research for possibility of autorefactoring of models
So I have .c4 input and I need .c4 as a result of processing
Is it possible?

**Comment by davydkov** [comment, marked as answer]
Hey @dzharikhin 

There is no printer/serializer at the moment. Though I think not difficult to add with `langium/generate`, like it is done [here](https://github.com/likec4/likec4/blob/main/packages/generators/src/mmd/generate-mmd.ts) but for mermaid.

Did you consider implementing refactorings as [code actions](https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/lsp/code-action.ts) inside existing language server?

**Comment by dzharikhin** [comment]
Thank you! I'll check both approaches
